docs/transactions/iti41.md

```diff
@@ -1,35 +1,228 @@
+    * Be [approved](https://profiles.ihe.net/ITI/TF/Volume3/ch-4.2.html#4.2.3.2.2), ie. `DocumentEntry.availabilityStatus = approved`, 
+    * Be valid, ie. the submission date must be within the [MTP `dosage`'s](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-dosage.html) `boundsPeriod` (`CHEMEDEPRDosage.repeat.bounds`).
+        * If the `boundsPeriod` is not defined, the entry is considered to be always active.
+        * If only a [`startDate`](http://hl7.org/fhir/R4/datatypes-definitions.html#Period.start) is specified, the MTP is considered to be active if the current date is after that date.
+    * Active, ie. `CHEMEDEPRDocumentMedicationTreatmentPlan.CHEMEDEPRMedicationStatement.status = active`
+    * For the same patient.
+ 
+* [PRE document's date of validity](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) (`CHEMEDEPRMedicationRequest.validityPeriod` ) must be after the [PRE document's creation time](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-composition-medicationprescription.html) (`CHEMEDCompositionMedicationPrescription.date`) and after the [start date of duration](https://build.fhir.org/ig/hl7ch/ch-emed//StructureDefinition-ch-emed-medicationstatement.html) (`CHEMEDMedicationStatement.effective[x].effectiveDateTime`) of treatment defined in the [MTP referenced by the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`).
+
+* Code of [medication](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) should match the code of medication of the [MTP referenced by the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-document-medicationprescription.html) (`CHEMEDEPRMedicationRequest.treatmentplan`).      
+* If [lot number](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medication.html) (`CHEMEDEPRMedication.batch`) is specified in referenced MTP, [the one in the PRE](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) (`CHEMEDEPRMedicationRequest.medication[x]:medicationReference.batch`) should match it.
+#### PADV
+##### PADV against a provisional PRE
+A "provisional PRE" is a prescription for which the [`CHEMEDEPRMedicationRequest.status`](https://build.fhir.org/ig/CARA-ch/ch-emed-epr/StructureDefinition-ch-emed-epr-medicationrequest.html) is `SUBMITTED` or `PROVISIONAL` and for which a PADV `OK` has not been completed yet. 
+* Medication cannot be changed.
+* Dosage instructions cannot be changed.
+* Substitution cannot be disallowed if the existing dispense includes one.
+* Observation code `REFUSE` cannot be issued.
+#####  PADV `CANCEL`
+* Not allowed against DIS or other PADV documents.
+
```

Comment:
Other implemented checks:

- PADV OK (sets targeted MTP or PRE status to ACTIVE):
  - against MTP: only allowed for treatment plans with status ACTIVE or SUSPENDED.
  - against PRE: only allowed for prescriptions with status SUBMITTED or PROVISIONAL
  - not allowed to target anything else than MTP or PRE
- PADV CHANGE (changes the treatment plan or prescription):
  - against MTP: 
    - the treatment status must be ACTIVE or SUSPENDED
    - the treatment must not have been prescribed
    - additionally, the treatment's status is set to ACTIVE as a result of the aggregation
  - against PRE:
    - the prescription's status must not be CANCELED or REFUSED
    - if the prescription's status is SUBMITTED, the aggregation sets it to ACTIVE as a result
  - not allowed to target anything else than MTP or PRE
- PADV CANCEL (sets status to CANCELED):
  - against MTP: 
    - sets also related prescriptions' status to CANCELED if their status was not already REFUSED or CANCELED.
    - treatment plan status must be ACTIVE or SUSPENDED
  - against PRE: not allowed for prescriptions whose status is already CANCELED or REFUSED. 
  - Not allowed to target anything else than MTP or PRE
- PADV REFUSE (sets status to REFUSED):
  - against MTP: 
    - sets also related prescriptions' status to REFUSED if the status was not already REFUSED or CANCELED.
    - treatment's status must be ACTIVE or SUSPENDED.
  - against PRE: the prescription's status must not be REFUSED, CANCELED or PROVISIONAL.
  - Not allowed to target anything else than MTP or PRE
- PADV SUSPEND (sets status to SUSPENDED): 
  - only allowed to target MTP entries
  - the treatment status must be ACTIVE
- PADV COMMENT: can target MTP, PRE and DIS entries only.